### PR TITLE
Fix homog phononic

### PR DESCRIPTION
**Pull request by vlukes**

### CODE REVIEW

File: sfepy/homogenization/coefs_phononic.py [hunk 1]
```diff
@@ -39,7 +39,9 @@ def compute_eigenmomenta(em_equation, var_name, problem, eig_vectors,
             eigenmomenta[ii, :] = 0.0
 
         else:
+            var.locked = False
             var.set_data(vec_phi.copy())
+            var.locked = True
```
Comment (rc) [inline sfepy/homogenization/coefs_phononic.py:44]: It might be better to use `variables.set_state_parts({var_name : vec_phi})`.

Comment (vlukes) [reply sfepy/homogenization/coefs_phononic.py]: Your fix has only one line of code, so it is definitely better :)

Comment (rc) [reply sfepy/homogenization/coefs_phononic.py]: But it does not work :] - try `python3 ./simple.py sfepy/examples/phononic/band_gaps.py --phonon-dispersion
`. One needs to call `variables.init_state()` at some time before calling `variables.set_state_parts()`.

The very original code works for me - you use it in some other example?

**Comment by rc** [comment]
Thanks, LGTM!

**Pull request merged**